resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventOutputImpl.java

```diff
@@ -182,25 +197,58 @@ private void internalFlushResponseToClient(boolean throwIOException) throws IOEx
 
             try
             {
+               CompletableFuture<Void> ret = new CompletableFuture<>();
                ServerResponseWriter.writeNomapResponse(jaxrsResponse, request, response,
                      ResteasyProviderFactory.getInstance(), t -> {
+                        AsyncOutputStream aos;
+                        try
+                        {
+                           aos = response.getAsyncOutputStream();
+                        } catch (IOException x)
+                        {
+                           close(false);
+                           ret.completeExceptionally(x);
+                           return;
+                        }
+                        // eager composition to guarantee ordering
+                        CompletionStage<Void> a = aos.asyncWrite(SseConstants.EOL);
```

Comment:
It does not really matter, but we could probably just add a new constant here to avoid the double async write.

Reply:
OK